pypeit/par/pypeitpar.py

```diff
@@ -1544,6 +1544,160 @@ def validate(self):
         # scipy.optimize.differential_evoluiton probalby checks this.
 
 
+class TellFitPar(ParSet):
+    """
+    A parameter set holding the arguments for sensitivity function computation using the UV algorithm, see
+    sensfunc.SensFuncUV
+
+    For a table with the current keywords, defaults, and descriptions,
+    see :ref:`pypeitpar`.
+    """
+
+    def __init__(self, objmodel=None, redshift=None, delta_redshift=None, pca_file=None, npca=None, bal_mask=None,
+                 bounds_norm=None, tell_norm_thresh=None, only_orders=None, pca_lower=None, pca_upper=None,
+                 star_type=None, star_mag=None, star_ra=None, star_dec=None, mask_abs_lines=None,
+                 func=None, model=None, polyorder=None, fit_region_mask=None, mask_lyman_a=None,
+                 delta_coeff_bounds=None, minmax_coeff_bounds=None, tell_grid=None):
+
+        # Grab the parameter names and values from the function arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = OrderedDict([(k, values[k]) for k in args[1:]])
+
+        # Initialize the other used specifications for this parameter set
+        defaults = OrderedDict.fromkeys(pars.keys())
+        dtypes = OrderedDict.fromkeys(pars.keys())
+        descr = OrderedDict.fromkeys(pars.keys())
+
+        defaults['tell_grid'] = None
+        dtypes['tell_grid'] = str
+        descr['tell_grid'] = 'pca pickle file. needed when you use qso_telluric'
+
+        defaults['only_orders'] = None
+        dtypes['only_orders'] = int
+        descr['only_orders'] = "order number if you only want to fit a single order"
+
+        defaults['objmodel'] = None
+        dtypes['objmodel'] = str
+        descr['objmodel'] = 'which object model you want to use for telluric fit'
+
+        defaults['redshift'] = 0.0
+        dtypes['redshift'] = [int, float]
+        descr['redshift'] = 'redshift for your object model'
+
+        ### Start parameters for qso_telluric
+        defaults['delta_redshift'] = 0.1
+        dtypes['delta_redshift'] = [int, float]
+        descr['delta_redshift'] = 'variable redshift range during the fit'
+
+        defaults['pca_file'] = os.path.join(resource_filename('pypeit', 'data/telluric/'),
+                                            'qso_pca_1200_3100.pckl')
+        dtypes['pca_file'] = str
+        descr['pca_file'] = 'pca pickle file. needed when you use qso_telluric'
+
+        defaults['npca'] = 8
+        dtypes['npca'] = int
+        descr['npca'] = 'Number of pca'
+
+        defaults['bal_mask'] = None
+        dtypes['bal_mask'] = [list, numpy.ndarray]
+        descr['bal_mask'] = 'List of bal masks '
+
+        defaults['bounds_norm'] = [0.1, 3.0]
+        dtypes['bounds_norm'] = list
+        descr['bounds_norm'] = "Normalization bounds"
```

Comment:
Can you add to these and the other descriptions?  These are pretty terse.
Thanks.

Reply:
done